main.js

```diff
@@ -0,0 +1,114 @@
+let BOOKS_DATA = [];
+
+function setStorage() {
+  localStorage.setItem('books_data', JSON.stringify(BOOKS_DATA));
+}
+
+function retrieveStorage() {
+  BOOKS_DATA = JSON.parse(localStorage.getItem('books_data'));
+}
```

Comment:
- [ ] [OPTIONAL] This logic can be moved to the top of this module and merged with the first line to prevent creating more boilerplate code in the module. Something like this:

```JavaScript
const BOOKS_DATA = JSON.parse(localStorage.getItem('books_data')) || [];
```

If you do that, you won't also need the following line of code:
```JavaScript
if (!localStorage.getItem('books_data')) {
  setStorage();
} else {
  retrieveStorage();
}
```
Because, if the app doesn't find anything in the localStorage you will always have an empty list to start off with. After the addition of the first data, you will set the result to localStorage and then the localStorage won't be null anymore. This way your module will be more precise.